src/datatypes/mod.rs

```diff
@@ -107,6 +107,8 @@ impl_daft_logical_datatype!(DateType, Date, Int32Type);
 impl_daft_logical_datatype!(TimeType, Unknown, Int64Type);
 impl_daft_logical_datatype!(DurationType, Unknown, Int64Type);
 impl_daft_logical_datatype!(EmbeddingType, Unknown, FixedSizeListType);
+impl_daft_logical_datatype!(ImageType, Unknown, ListType);
```

Comment:
Shouldn't the non-fixed-size Image be backed by a struct to also encode `H, W, etc`,

Reply:
Yep I'm working on that right now, the casting is the main blocker so I kept it was a `ListType` in order to get everything working e2e without reconstructing the image shape on egress, but forgot to go back and update it. Working on that now.